envi/config.py

```diff
@@ -80,6 +80,48 @@ def getOptionDoc(self, optname):
         '''
         return self.cfgdocs.get(optname)
 
+    def getConfigPaths(self, with_vals=False):
+        '''
+        Return a list of valid path strings
+        If "with_vals" is True, also include the current values
+        '''
+        paths = []
+        todo = [ ([], self) ]
+
+        while todo:
+            path, config = todo.pop()
+            
+            if config.keys():
+                pathstr = '.'.join(path) + "."
+                if with_vals:
+                    newpaths = [ "%s%s = %s" % (pathstr, key, config[key])  for key in config.keys()]
```

Comment:
A programatic "getter" API like this should be returning structured data not display content.  How about we return a list of (path,valu) tuples so other code which isn't just interested in humon readable strings wont have to implement the same logic (for example, a GUI element).  Then let the "repr" version below create display strings from the tuples.